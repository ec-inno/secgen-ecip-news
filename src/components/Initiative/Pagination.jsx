import React from 'react';

const Pagination = ({ onClick }) => (
  <div className="ecl-row ecl-u-mt-l">
    <div className="ecl-col-sm-12 ecl-col-md-12">
      <nav className="ecl-pagination" aria-label="Pagination">
        <ul className="ecl-pagination__list">
          <li className="ecl-pagination__item ecl-pagination__item--next">
            <a
              onClick={onClick}
              aria-label="Go to next page"
              href="#"
              className="ecl-pagination__link ecl-link ecl-link--standalone ecl-link--icon ecl-link--icon-after"
            >
              <span className="ecl-link__label">See more initiatives</span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
);

export default Pagination;
