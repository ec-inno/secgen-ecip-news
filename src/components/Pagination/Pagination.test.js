import React from 'react';
import renderer from 'react-test-renderer';

import Pagination from './Pagination';

describe('Pagination', () => {
  it('Returns null on empty input', () => {
    const tree = renderer.create(<Pagination />).toJSON();
    expect(tree).toBe(null);
  });

  it('Renders list items', () => {
    const props = {
      label: 'test',
      items: [{ label: '1', href: '#1', label: 'Page 1' }],
    };
    const tree = renderer.create(<Pagination {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
