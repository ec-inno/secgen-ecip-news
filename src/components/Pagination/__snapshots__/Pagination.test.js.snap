// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Renders list items 1`] = `
<nav
  aria-label="test"
  className="ecl-pagination"
>
  <ul
    className="ecl-pagination__list"
  >
    <li
      className="ecl-pagination__item"
    >
      <span
        className="ecl-pagination__text ecl-pagination__text--summary"
      >
        Page 1
      </span>
      <span
        className="ecl-pagination__text ecl-pagination__text--full"
      />
    </li>
  </ul>
</nav>
`;
